internal/provider/provider_config.go

```diff
@@ -37,6 +40,49 @@ func (c *apiClient) loadAndValidate(ctx context.Context) diag.Diagnostics {
 		c.ClientScopes = DefaultClientScopes
 	}
 
+	if c.AccessToken != "" {
+		contents, _, err := pathOrContents(c.AccessToken)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		token := &oauth2.Token{AccessToken: contents}
+
+		log.Printf("[INFO] Authenticating using configured Google JSON 'access_token'...")
+		log.Printf("[INFO]   -- Scopes: %s", c.ClientScopes)
+
+		if c.ImpersonatedUserEmail != "" {
+
+			if c.ServiceAccount == "" {
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  "service_account is required to impersonate a user with the access_token authentication.",
+				})
+
+				return diags
+			}
+
+			tokenSource, err := impersonate.CredentialsTokenSource(context.TODO(), impersonate.CredentialsConfig{
+				TargetPrincipal: c.ServiceAccount,
+				Scopes:          c.ClientScopes,
+				Subject:         c.ImpersonatedUserEmail,
+			}, option.WithTokenSource(oauth2.StaticTokenSource(token)))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			creds := googleoauth.Credentials{
+				TokenSource: tokenSource,
+			}
+			diags = c.SetupClient(ctx, &creds)
+			return diags
+		}
+
+		creds := googleoauth.Credentials{
+			TokenSource: oauth2.StaticTokenSource(token),
+		}
+		diags = c.SetupClient(ctx, &creds)
```

Comment:
Hi, sorry for adding this late, but isn't a `return` statement missing here? This way, if there are no impersonated user specified, it will fall back to ADC at line 109, so the access token will be ignored finally.